src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,226 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user1;
+    private User user2;
+    private User user3;
+
+    @BeforeAll
+    static void setUpAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        // Technically valid users.
+        // Three of them for testing edge cases, because duplicates are not allowed to store.
+        // Clearing static storage list before each test.
+        user1 = new User();
+        user1.setLogin("uniqueLogin1");
+        user1.setPassword("password");
+        user1.setAge(20);
+
+        user2 = new User();
+        user2.setLogin("uniqueLogin2");
+        user2.setPassword("password");
+        user2.setAge(20);
+
+        user3 = new User();
+        user3.setLogin("uniqueLogin3");
+        user3.setPassword("password");
+        user3.setAge(20);
+
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user1.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+
+        boolean containsNone = Storage.people.isEmpty();
```

Comment:
In general, tests are correct, but you might want to learn about the convention that is being globally used for writing unit tests: grouping code in given/when/then labels, which makes them easier to understand (you clearly see distinguish between input data, test subject and results validation).

Here's an example description: https://javastart.pl/baza-wiedzy/testowanie-jednostkowe/given-when-then-w-testach